app_dart/lib/src/request_handlers/get_build_status.dart

```diff
@@ -32,6 +44,9 @@ base class GetBuildStatus extends RequestHandler<Body> {
     final repoName = request!.uri.queryParameters[_kRepoParam] ?? 'flutter';
     final slug = RepositorySlug('flutter', repoName);
     final status = (await _buildStatusService.calculateCumulativeStatus(slug))!;
+
+    await maybeNotifyDiscord(status);
```

Comment:
So this service is called a *lot*, like, it's called every single time someone requests a badge of the build status, every time the dashboard refreshes, etc.

A better implementation would be a cron-based implementation (every 5m) that is something like `TreeStatusMessageUpdater extends GetBuildStatus`.

See the request handler in this same directory that does that for SVG badge generator for inspiration. It also means we would not have to fake out/provide an HTTP client to every `GetBuildStatus`.

Reply:
This is already called a lot - but it's wrapped in a cache service and rate limited to once every 15 seconds, so on the grand-scale of things, not really that often (0.07 qps).   The http client is only provided if the status has changed.

Reply:
Moved